internal/controller/nim/account_controller.go

```diff
@@ -383,6 +440,52 @@ func (r *AccountReconciler) reconcileNimPullSecret(
 	return secret, nil
 }
 
+// getSelectedModelList returns a list of Ids of the selected models
+func (r *AccountReconciler) getSelectedModelList(
+	ctx context.Context, cmRef *corev1.ObjectReference,
+	namespace string) ([]string, error) {
+	logger := log.FromContext(ctx)
+	logger.V(1).Info("getting selected model list")
+
+	// if selected model list is not set
+	if cmRef == nil || len(cmRef.Name) == 0 {
+		return []string{}, nil
+	}
+
+	// get the config map that contains the selected model list
+	cmNs := cmRef.Namespace
+	if cmNs == "" {
+		cmNs = namespace
+	}
+	modelListCm := &corev1.ConfigMap{}
+	modelListCmSubject := types.NamespacedName{Name: cmRef.Name, Namespace: cmNs}
+	if err := r.Client.Get(ctx, modelListCmSubject, modelListCm); err != nil {
+		if k8serrors.IsNotFound(err) {
+			logger.Error(err, "failed to fetch the config map for getting the selected model list",
+				"config map", modelListCmSubject)
+			return []string{}, nil
+		}
+		return nil, err
+	}
+
+	// convert the selected model list to a string array
+	if modelListCm.Data != nil {
+		if models, ok := modelListCm.Data["models"]; ok {
+			var selectedModelIds []string
+			if err := json.Unmarshal([]byte(models), &selectedModelIds); err != nil {
```

Comment:
If we use a comma-seperated string, should we still unmarshall?

Reply:
I am still using json array. 
But I don't use json marshal to create the logs any more, to avoid additional complexity as you mentioned in the comments.

IMHO, it is easier to edit 100 elements in a json array with a json editor, than editing a long comma delimited string.

Reply:
Gotcha.